lib/puppet/util/package/version/gem.rb

```diff
@@ -0,0 +1,15 @@
+module Puppet::Util::Package::Version
+  class Gem < ::Gem::Version
+    def self.parse(version)
+      raise ValidationFailure, version.to_s unless version.is_a? String
```

Comment:
no need to use `to_s` as you interpolate in the` initialize` of `ValidationFailure`